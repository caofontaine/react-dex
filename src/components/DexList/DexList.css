table {
  border-radius: 10px;
  border-collapse: collapse;
  width: 100%;
  -webkit-box-shadow: 0 0 5px 6px rgba(0, 0, 0, 0.4);
  -moz-box-shadow:    0 0 5px 6px rgba(0, 0, 0, 0.4);
  box-shadow:         0 0 5px 6px rgba(0, 0, 0, 0.4);
  margin: 1%;
}

table, tr {
  border: 2px solid #000;
}
