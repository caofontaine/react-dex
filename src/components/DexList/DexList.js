import React from 'react';
import DexRow from '../DexRow/DexRow';
import './DexList.css';

const DexList = ({ dex }) => {
	return (

      <table>
				<thead>
	        <tr>
	          <th>#</th>
	          <th>&nbsp;</th>
	          <th>Name</th>
	          <th>Type 1</th>
	          <th>Type 2</th>
	          <th>Shiny</th>
	        </tr>
				</thead>

				<tbody>
					{
						dex.map(pokemon => {
							return (
								<DexRow
									key={pokemon.dexnum + '*'}
									dexNum={pokemon.dexnum}
									name={pokemon.name}
									type1={pokemon.type1}
									type2={pokemon.type2}
									region={pokemon.region}
								/>
							)
						})
					}
				</tbody>
      </table>

	)
}

export default DexList;
