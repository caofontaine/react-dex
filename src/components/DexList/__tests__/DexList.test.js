import React from 'react';
import { shallow } from 'enzyme';
import DexList from '../DexList';

it('Expect to render DexList component', () => {
  const mockDex = [
    {
      dexNum: '0004',
      name: 'Charmander',
      type1: 'Fire',
      type2: null,
      region: 'Kanto'
    }
  ];
  expect(shallow(<DexList dex={mockDex} />)).toMatchSnapshot();
});
