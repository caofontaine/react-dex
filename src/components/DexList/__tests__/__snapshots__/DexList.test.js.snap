// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render DexList component 1`] = `ShallowWrapper {}`;
